lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Azure.DataLake.Store;
+using Microsoft.Azure.DataLake.Store.FileTransfer;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureDataLake
+{
+    /// <summary>
+    /// An IFileSystem implementation for Azure Data Lake Store.
+    /// </summary>
+    internal sealed class AzureDataLakeFileSystem : IFileSystem
+    {
+        IDataLakeStoreClient _client;
+
+        [Inject]
+        private AzureDataLakeFileSystem(IDataLakeStoreClient client)
```

Comment:
This is pretty good but a little round about. Let me see if there is a cleaner way and I'll get back to you.

Reply:
I think using Tangs IExternalConstructor would be more appropriate. Let's discuss in person.

Reply:
As per offline discussion, going with current implementation as it was followed in AzureBlockBlobFileSystem